sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Extensions/WebPubSubRequestExtensions.cs

```diff
@@ -199,9 +197,11 @@ internal static bool IsValidSignature(this WebPubSubConnectionContext connection
             return null;
         }
 
-        internal static string EncodeConnectionStates(this Dictionary<string, object> value)
+        internal static string EncodeConnectionStates(this IReadOnlyDictionary<string, BinaryData> value)
         {
-            return Convert.ToBase64String(Encoding.UTF8.GetBytes(JsonSerializer.Serialize(value)));
```

Comment:
Here we encode the connection states into response header, if we’d like to support ‘setState(key, BinaryData.FromBinary)”, we should not use BinaryData.ToString in the converter